plugins/inputs/beat/beat.go

```diff
@@ -0,0 +1,268 @@
+	Libbeat  interface{}            `json:"libbeat"`
+	System   interface{}            `json:"system"`
+}
+
+type Beat struct {
+	URL string `toml:"url"`
+
+	CollectBeatStats     *bool `toml:"collect_beat_stats"`
+	CollectLibbeatStats  *bool `toml:"collect_libbeat_stats"`
+	CollectSystemStats   *bool `toml:"collect_system_stats"`
+	CollectFilebeatStats *bool `toml:"collect_filebeat_stats"`
+
+	Username   string            `toml:"username"`
+	Password   string            `toml:"password"`
+	Method     string            `toml:"method"`
+	Headers    map[string]string `toml:"headers"`
+	HostHeader string            `toml:"host_header"`
+	Timeout    internal.Duration `toml:"timeout"`
+
+	tls.ClientConfig
+	client *http.Client
+}
+
+const (
+	defaultURL    = "http://127.0.0.1:5066"
+	defaultMethod = "GET"
+)
+
+var defaultCollectBeatStats = true
+var defaultCollectLibbeatStats = true
+var defaultCollectSystemStats = false
+var defaultCollectFilebeatStats = true
+var defaultTimeout = internal.Duration{Duration: time.Second * 5}
+
+func (beat *Beat) Init() error {
+	var err error
+	beat.client, err = beat.createHTTPClient()
+
+	if err != nil {
+		return err
+	}
+	if beat.URL == "" {
+		beat.URL = defaultURL
+	}
+	if beat.CollectBeatStats == nil {
+		beat.CollectBeatStats = &defaultCollectBeatStats
+	}
+	if beat.CollectLibbeatStats == nil {
+		beat.CollectLibbeatStats = &defaultCollectLibbeatStats
+	}
+	if beat.CollectSystemStats == nil {
+		beat.CollectSystemStats = &defaultCollectSystemStats
+	}
+	if beat.CollectFilebeatStats == nil {
+		beat.CollectFilebeatStats = &defaultCollectFilebeatStats
+	}
+	if beat.Method == "" {
+		beat.Method = defaultMethod
+	}
+	if beat.Headers == nil {
+		beat.Headers = make(map[string]string)
+	}
+	if beat.Timeout.Duration == 0 {
+		beat.Timeout = defaultTimeout
+	}
+	return nil
+}
+
+func (beat *Beat) Description() string {
+	return description
+}
+
+func (beat *Beat) SampleConfig() string {
+	return sampleConfig
+}
+
+// createHTTPClient create a clients to access API
+func (beat *Beat) createHTTPClient() (*http.Client, error) {
+	tlsConfig, err := beat.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: beat.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+// gatherJSONData query the data source and parse the response JSON
+func (beat *Beat) gatherJSONData(url string, value interface{}) error {
+	request, err := http.NewRequest(beat.Method, url, nil)
+	if err != nil {
+		return err
+	}
+
+	if beat.Username != "" {
+		request.SetBasicAuth(beat.Username, beat.Password)
+	}
+	for k, v := range beat.Headers {
+		request.Header.Add(k, v)
+	}
+	if beat.HostHeader != "" {
+		request.Host = beat.HostHeader
+	}
+
+	response, err := beat.client.Do(request)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	err = json.NewDecoder(response.Body).Decode(value)
+	return err
```

Comment:
You can also do
```suggestion
	return json.NewDecoder(response.Body).Decode(value)
```
Both ok for me.